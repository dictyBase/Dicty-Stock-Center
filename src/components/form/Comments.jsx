import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from 'rebass'
import { ControlLabel, FormGroup, FormTextArea } from 'styles'

export default class Comments extends Component {
  displayName = 'form contact information'

  static propTypes = {
      comments: PropTypes.object.isRequired,
      placeholder: PropTypes.string,
      rows: PropTypes.string
  }

  render() {
      const { comments, placeholder, rows, children } = this.props
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ [1, 1 / 4, 1 / 4] } ml={ 1 } mb={ 2 } mr={ 1 }>
            <ControlLabel>{ children }</ControlLabel>
          </Box>
          <Box w={ [1, '60%', '60%'] } mr={ 1 } ml={ 1 }>
            <FormTextArea
              rows={ rows }
              {...comments}
              value={ comments.value }
              placeholder={ placeholder }
            />
          </Box>
        </Flex>
      </FormGroup>
    )
  }
}
