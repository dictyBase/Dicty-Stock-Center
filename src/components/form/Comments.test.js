import React from 'react'
import renderer from 'react-test-renderer'
import Comments from './Comments'

test('matching a snapshot of Comments', () => {
    const comments = {
        value: undefined
    }
    const placeholder = 'placeholder'
    const rows = '20'

    const component = renderer.create(
    <Comments
        rows={ rows }
        comments={ comments }
        placeholder={ placeholder }
    />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
