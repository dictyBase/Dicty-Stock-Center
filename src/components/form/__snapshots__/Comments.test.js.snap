// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Comments 1`] = `
<div
  className="sc-kLIISr itrVsn"
>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
      is={null}
    >
      <label
        className="sc-dliRfk eLcvEu"
      />
    </div>
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
      is={null}
    >
      <textarea
        className="sc-iGrrsa lcsVhf"
        placeholder="placeholder"
        rows="20"
        value={undefined}
      />
    </div>
  </div>
</div>
`;
